File: Day225/leftAndRightMostNodesOfBT.go

package main

import (
	"fmt"
	"github.com/golang-collections/collections/queue"
)

type Node struct {
	data  int
	left  *Node
	right *Node
}

func printCorner(node *Node) {
	q := queue.New()
	q.Enqueue(node)
	for !q.Empty() {
		size := q.Len()
		for i := 0; i < size; i++ {
			temp := q.Peek().(*Node)
			q.Dequeue()
			if i == 0 || i == size-1 {
				fmt.Printf("%d ", temp.data)
			}
			if temp.left != nil {
				q.Enqueue(temp.left)
			}
			if temp.right != nil {
				q.Enqueue(temp.right)
			}
		}
	}
}

File: Day225/leftAndRightMostNodesOfBT.java
class Tree
{
    public static void printCorner(Node node)
    {
        Queue<Node> queue = new LinkedList<>();
    queue.offer(node);
    while (!queue.isEmpty()) {
        int size = queue.size();
        for (int i=0; i<size; i++) {
            Node temp = queue.peek();
            queue.poll();
            if (i == 0) {
                System.out.print(temp.data + " ");
            } else if (i == size - 1) {
                System.out.print(temp.data + " ");
            }
            if (temp.left != null) {
                queue.offer(temp.left);
            }
            if (temp.right != null) {
                queue.offer(temp.right);
            }
        }
    }
        // add your code here    
    }
    
}
